internal/services/applications/application_resource.go

```diff
@@ -57,21 +53,101 @@ func applicationResource() *schema.Resource {
 				Type:             schema.TypeString,
 				Optional:         true,
 				Computed:         true,
-				Deprecated:       "This property has been renamed to `display_name` and will be removed in version 2.0 of this provider.",
+				Deprecated:       "This property has been renamed to `display_name` and will be removed in version 2.0 of the AzureAD provider",
 				ExactlyOneOf:     []string{"display_name", "name"},
 				ValidateDiagFunc: validate.NoEmptyStrings,
 			},
 
+			"api": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						// TODO: v2.0 also consider another computed typemap attribute `oauth2_permission_scope_ids` for easier consumption
+						"oauth2_permission_scope": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Computed: true,
+							Set:      tf.SetFuncHashId,
```

Comment:
(as below) the set function only using the ID will only track changes to the ID, which likely isn't what we want?